controllers/base_controller.js

```diff
@@ -116,54 +132,76 @@ module.exports = function BaseControllerModule(pb) {
         this.pathVars            = props.path_vars;
         this.query               = props.query;
         this.pageName            = '';
+        this.site                = props.site;
 
-        var self   = this;
         var tsOpts = {
             ls: this.localizationService,
-            activeTheme: props.activeTheme
+            activeTheme: props.activeTheme,
+            site: this.site
         };
-        this.templateService = new pb.TemplateService(tsOpts);
-        this.templateService.registerLocal('locale', this.ls.language);
-        this.templateService.registerLocal('error_success', function(flag, cb) {
+
+
+        this.ts = this.getTemplateService(tsOpts);
+        this.ts.registerLocal('locale', this.ls.language);
+        this.ts.registerLocal('error_success', function(flag, cb) {
             self.displayErrorOrSuccessCallback(flag, cb);
         });
-        this.templateService.registerLocal('page_name', function(flag, cb) {
+        this.ts.registerLocal('page_name', function(flag, cb) {
             cb(null, self.getPageName());
         });
-        this.templateService.registerLocal('localization_script', function(flag, cb) {
+        this.ts.registerLocal('localization_script', function(flag, cb) {
             self.requiresClientLocalizationCallback(flag, cb);
         });
-        this.templateService.registerLocal('analytics', function(flag, cb) {
+        this.ts.registerLocal('analytics', function(flag, cb) {
             pb.AnalyticsManager.onPageRender(self.req, self.session, self.ls, cb);
         });
-        this.templateService.registerLocal('wysiwyg', function(flag, cb) {
+        this.ts.registerLocal('wysiwyg', function(flag, cb) {
             var wysiwygId = util.uniqueId();
 
-            self.templateService.registerLocal('wys_id', wysiwygId);
-            self.templateService.load('admin/elements/wysiwyg', function(err, data) {
+            self.ts.registerLocal('wys_id', wysiwygId);
+            self.ts.load('admin/elements/wysiwyg', function(err, data) {
                 cb(err, new pb.TemplateValue(data, false));
             });
         });
-        this.ts = this.templateService;
-        
+
+
         /**
          *
          * @property activeTheme
          * @type {String}
          */
         this.activeTheme = props.activeTheme;
-        
-        //build out a base service context that can be cloned and passed to any 
+
+        //build out a base service context that can be cloned and passed to any
         //service objects
         this.context = {
             req: this.req,
             session: this.session,
             ls: this.ls,
             ts: this.ts,
-            activeTheme: this.activeTheme
+            site: this.site,
+            activeTheme: this.activeTheme,
+            onlyThisSite: true
         };
 
-        cb();
+        var siteService = new SiteService();
+        siteService.getByUid(this.site, function (err, siteInfo) {
```

Comment:
Added all site info required into the site_map of request handler.  The DB request has been removed.

Ben Tidwell
Software Engineer III | Talent Network
CareerBuilder
5550-A Peachtree Pkwy, Norcross GA 30092

From: Brian <notifications@github.com<mailto:notifications@github.com>>
Reply-To: pencilblue/pencilblue <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Thursday, July 9, 2015 at 5:41 PM
To: pencilblue/pencilblue <pencilblue@noreply.github.com<mailto:pencilblue@noreply.github.com>>
Cc: Ben Tidwell <ben.tidwell@careerbuilder.com<mailto:ben.tidwell@careerbuilder.com>>
Subject: Re: [pencilblue] Multi Tenancy (#680)


In controllers/base_controller.js<https://github.com/pencilblue/pencilblue/pull/680#discussion_r34307100>:

>          };
>
> -        cb();
> +        var siteService = new SiteService();
> +        siteService.getByUid(this.site, function (err, siteInfo) {


It is extremely inefficient to have to fire off a DB call for EVERY request. The site service should provide some kind of caching mechanism to avoid the extra DB call.

There are a few ways to accomplish this:
The easiest would be to use the simple layered service along with a configuration block to set the level of caching that can be used.

At scale or under load this is really going hurt. It is a simple optimization to save a shit ton of resources over time.

-
Reply to this email directly or view it on GitHub<https://github.com/pencilblue/pencilblue/pull/680/files#r34307100>.